dataikuapi/dss/codestudio.py

```diff
@@ -0,0 +1,141 @@
+    def __init__(self, client, project_key, data):
+        self.client = client
+        self.project_key = project_key
+        self._data = data
+
+    def to_code_studio_object(self):
+        """Gets the :class:`DSSCodeStudioObject` corresponding to this code studio object """
+        return DSSCodeStudioObject(self.client, self.project_key, self._data["id"])
+
+    @property
+    def name(self):
+        return self._data["name"]
+    @property
+    def id(self):
+        return self._data["id"]
+    @property
+    def owner(self):
+        return self._data["owner"]
+    @property
+    def template_id(self):
+        return self._data["templateId"]
+    @property
+    def template_label(self):
+        return self._data.get('desc', {}).get('label', self._data['id'])
+    @property
+    def template_description(self):
+        return self._data.get('desc', {}).get('shortDesc', '')
+
+
+class DSSCodeStudioObject(object):
+    """
+    A handle to manage a code studio object of a project
+    """
+    def __init__(self, client, project_key, code_studio_object_id):
+        """Do not call directly, use :meth:`dataikuapi.dss.project.DSSProject.get_code_studio_object`"""
+        self.client = client
+        self.project_key = project_key
+        self.code_studio_object_id = code_studio_object_id
+
+    def delete(self):
+        """
+        Delete the code studio
+        """
+        self.client._perform_empty("DELETE", "/projects/%s/code-studios/%s" % (self.project_key, self.code_studio_object_id))
+
+    def get_settings(self):
+        """
+        Get the code studio object's settings
+
+        :returns: a handle to manage the code studio settings
+        :rtype: :class:`dataikuapi.dss.codestudio.DSSCodeStudioObjectSettings`
+        """
+        settings = self.client._perform_json("GET", "/projects/%s/code-studios/%s" % (self.project_key, self.code_studio_object_id))
+        return DSSCodeStudioObjectSettings(self.client, self.project_key, self.code_studio_object_id, settings)
+
+    def get_status(self):
+        """
+        Get the code studio object's state
+
+        :returns: a handle to inspect the code studio state
+        :rtype: :class:`dataikuapi.dss.codestudio.DSSCodeStudioObjectStatus`
+        """
+        status = self.client._perform_json("GET", "/projects/%s/code-studios/%s/status" % (self.project_key, self.code_studio_object_id))
+        return DSSCodeStudioObjectStatus(self.client, self.project_key, self.code_studio_object_id, status)
+
+    def stop(self):
+        """
+        Stop a running code studio
+
+        :returns: a future to wait on the stop, or None if already stopped
+        :rtype: :class:`dataikuapi.dss.future.DSSFuture`
+        """
+        ret = self.client._perform_json("POST", "/projects/%s/code-studios/%s/stop" % (self.project_key, self.code_studio_object_id))
+        if 'jobId' in ret:
+            return self.client.get_future(ret["jobId"])
+        else:
```

Comment:
Should not return null if not job id, but an already completed future. Use DSSFuture.from_resp